test/controllers/devise_token_auth/registrations_controller_test.rb

```diff
@@ -81,6 +81,33 @@ class DeviseTokenAuth::RegistrationsControllerTest < ActionDispatch::Integration
       end
     end
 
+    describe 'using allow_unconfirmed_access_for' do
+      before do
+        @original_duration = Devise.allow_unconfirmed_access_for
+        Devise.allow_unconfirmed_access_for = nil
```

Comment:
if this is null, how can it return the headers?

Reply:
Setting this to nil means unlimited time period for accessing without confirmation.

See [Devise Docs](http://www.rubydoc.info/github/plataformatec/devise/master/Devise%2FModels%2FConfirmable:confirmation_period_valid%3F)

Reply:
That's the solution that posted [Zach in the issue](https://github.com/lynndylanhurley/devise_token_auth/issues/89#issuecomment-333309952). In fact, I cloned your branch, commented `active_for_authentication?` and the tests still pass, so I don't see any improvement or your tests are wrong.

Reply:
Commenting out the active_for_authentication? is not the same as reverting the code change since there was previously an if \ else which is now two separate ifs.

If you are saying that there is no test to make sure that we don't send when the headers when the user is not confirmed I can write a test for that as well. I assumed that there was an existing test covering that case but I suppose that is not the case.

I discovered this issue because I was not receiving the headers on the successful registration.

What Zach posted in the issue is correct, the registration succeeds and the user can subsequently log in. However the auth headers are definitely currently not returned from the registration.

Reply:
Thank you for approving.